pkg/vsphere/disk/disk.go

```diff
@@ -45,6 +45,40 @@ type Filesystem interface {
 	Unmount(path string) error
 }
 
+type ReferenceCount struct {
+	owner      string
+	refname    string
+	references int
+}
+
+func NewReferenceCount(o, n string) *ReferenceCount {
+	return &ReferenceCount{
+		owner:      o,
+		refname:    n,
+		references: 0,
+	}
+}
+
+func (r *ReferenceCount) Increment() {
```

Comment:
See below.. atomics are used in places where the disk needs to be locked, and these counters are wrapped in those locks.